src/deadlineresolver.js

```diff
@@ -208,7 +208,7 @@ class DeadlineResolver {
       debug('Resolved taskId: %s, by deadline', taskId);
 
       // Update dependency tracker
-      await this.dependencyTracker.resolveTask(taskId, 'exception');
+      await this.dependencyTracker.resolveTask(taskId, task.taskGroupId, 'exception');
```

Comment:
Further up in this file... if `if (!task) {` and we return immediately, we need to remove any partially constructed task...

Basically, we'll have to call `this.TaskGroupActiveSet.remove({taskId, taskGroupId}, true);` and check for emptiness and send pulse message if empty...

The `if (!task` case happens if the task wasn't created by the deadline message was sent... Due to a crash... In that case the `TaskGroupActiveSet` entry might have been created... We should make sure it's cleaned up... And that the pulse message is sent...

So failing to retry a createTask (which is what can cause this), will result in a task-group something that's resolved by deadline... I can live with that... Obviously, that's an extremely delayed message, but a lot better than nothing.